apps/wallet-mobile/src/yoroi-wallets/cardano/api/metadata.ts

```diff
@@ -22,11 +22,11 @@ export const getNFTs = async (ids: string[], config: BackendConfig): Promise<Bal
 
   const [assetMetadatas, assetSupplies] = await Promise.all([
     fetchDefault<unknown>('multiAsset/metadata', payload, config),
-    fetchTokensSupplies(ids, config),
+    fetchTokensSupplies(ids as Array<`${string}.${string}`>, config),
```

Comment:
there is a loop that maps from ids (Balance) to assets {k,v}, I think you can use the same logic to transform from Balance Token Id, to fetchToeknsSupplies id with a type guard, --> `filter(isTypeguard)` by filtering out if is not

Reply:
@MichalSzorad  The Api introduced the type change to include the dot. Either `getNFTs` uses that type as input as well, or it gets casted like you did here, or inside `fetchTokensSupplies`. Not very type safe, and not actually handling anything, but it silences the compiler.

If we want to move the type up to the input, then I'd alias it and force it everywhere.
```
type TokenId = `${string}.${string}`
```

Problem is, forcing this pattern from strings on usage will get tedious fast.

The other "safe" alternative is to add, either here or inside `fetchTokensSupplies`, an actual runtime check to make sure the string has the dot, or throw otherwise, then handle the throw possibility.

Reply:
The `filter(isTypeguard)` would silently ignore wrong ones, could lead to confusion maybe. Not sure how critical this is.

Reply:
I see there's utils around this, you could map the ids with `toTokenId`.
It doesn't really handle incorrect strings other than getting a wrong result, but it will cast the types for you.

Reply:
casting is flaky, it wouldn't prevent the API of returning error if the payload is safely parsed on the other side (if it has validators) so for 50 to 500 assets that you can consume at once, if you provide a single one (i.e primary token) as input the entire request would fail. By filtering, and not casting, we are saying that we don't mind about those that are not matching the expected payload type of the API, cuz the API doesn't support it. I believe it boils down to, shall we throw before even firing the request and client deal with it, or as we know what is supported by the API, we get the data that we can? I'd prefer the 2nd one.